src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -73,93 +80,104 @@ public GlobbingUrlBuilder(IFileProvider fileProvider, IMemoryCache cache, PathSt
-                IEnumerable<string> files;
-                if (!Cache.TryGetValue(cacheKey, out files))
+                var includeTokenizer = new StringTokenizer(include, ',');
+                var includeEnumerator = includeTokenizer.GetEnumerator();
+                if (!includeEnumerator.MoveNext())
                 {
-                    var options = new MemoryCacheEntryOptions();
-
-                    for (var i = 0; i < includePatterns.Length; i++)
-                    {
-                        var changeToken = FileProvider.Watch(includePatterns[i]);
-                        options.AddExpirationToken(changeToken);
-                    }
-
-                    files = FindFiles(includePatterns, excludePatterns);
+                    return EmptyList;
+                }
 
-                    Cache.Set(cacheKey, files, options);
+                var options = new MemoryCacheEntryOptions();
+                var trimmedIncludePatterns = new List<string>();
+                foreach (var includePattern in includeTokenizer)
+                {
+                    var changeToken = FileProvider.Watch(includePattern.Value);
+                    options.AddExpirationToken(changeToken);
+                    trimmedIncludePatterns.Add(TrimLeadingTildeSlash(includePattern));
                 }
-                return files;
+
+                files = Cache.Set<List<string>>(
+                    cacheKey,
+                    FindFiles(trimmedIncludePatterns, exclude),
+                    options);
             }
 
-            return FindFiles(includePatterns, excludePatterns);
+            return files;
         }
 
-        private IEnumerable<string> FindFiles(string[] includePatterns, string[] excludePatterns)
+        private List<string> FindFiles(List<string> trimmedIncludePatterns, string exclude)
         {
             var matcher = MatcherBuilder != null ? MatcherBuilder() : new Matcher();
-            var trimmedIncludePatterns = new List<string>();
-            for (var i = 0; i < includePatterns.Length; i++)
-            {
-                trimmedIncludePatterns.Add(TrimLeadingTildeSlash(includePatterns[i]));
-            }
             matcher.AddIncludePatterns(trimmedIncludePatterns);
 
-            if (excludePatterns != null)
+            if (!string.IsNullOrWhiteSpace(exclude))
             {
+                var excludeTokenizer = new StringTokenizer(exclude, ',');
                 var trimmedExcludePatterns = new List<string>();
-                for (var i = 0; i < excludePatterns.Length; i++)
+                foreach (var excludePattern in excludeTokenizer)
                 {
-                    trimmedExcludePatterns.Add(TrimLeadingTildeSlash(excludePatterns[i]));
+                    trimmedExcludePatterns.Add(TrimLeadingTildeSlash(excludePattern));
                 }
                 matcher.AddExcludePatterns(trimmedExcludePatterns);
             }
```

Comment:
Probably more readable if you
- factor out this block into a separate method for readability
- do the `exclude` parsing in `ExpandGlobbedUrl()` for consistency